src/widgets/dialogs/EmotePopup.cpp

```diff
@@ -137,40 +161,62 @@ EmotePopup::EmotePopup(QWidget *parent)
     auto layout = new QVBoxLayout(this);
     this->getLayoutContainer()->setLayout(layout);
 
-    this->notebook_ = new Notebook(this);
-    layout->addWidget(this->notebook_);
-    layout->setMargin(0);
+    QRegExp searchRegex("\\S*");
+    searchRegex.setCaseSensitivity(Qt::CaseInsensitive);
+    QValidator *searchValidator = new QRegExpValidator(searchRegex);
+
+    this->search_ = new QLineEdit();
+    this->search_->setPlaceholderText("Search all emotes...");
+    this->search_->setValidator(searchValidator);
+    this->search_->setClearButtonEnabled(true);
+    this->search_->findChild<QAbstractButton *>()->setIcon(
```

Comment:
Missing include of QAbstractButton in this file, add it to the top

`#include <QAbstractButton>`